src/viewers/VivViewer.js

```diff
@@ -78,13 +89,55 @@ export default class VivViewer extends PureComponent {
     return viewState;
   }
 
+  componentDidUpdate(prevProps) {
+    const { props } = this;
+    const { views } = props;
+    // Only update state if the previous viewState prop does not match the current one
+    // so that people can update viewState
+    const changed = views
+      .filter(view => {
+        const prevViewState = prevProps.viewStates.find(
+          viewState => viewState.id === view.id
+        );
+        const currViewState = props.viewStates.find(
+          viewState => viewState.id === view.id
+        );
+        return (
+          currViewState && !areViewStatesEqual(currViewState, prevViewState)
+        );
+      })
+      .map(view => view.id);
+    if (changed.length) {
+      const { viewStates: prevStates } = this.state;
+      const viewStates = {};
+      views.forEach(view => {
+        if (changed.includes(view.id)) {
+          const { height, width } = view;
+          const viewState = props.viewStates.find(v => v.id === view.id);
```

Comment:
Re https://github.com/hms-dbmi/viv/pull/359/files#r561125651:
```javascript
const viewState = props.viewStates.find(v => v.id === view.id);
if (changed.includes(view.id) && viewState) {
  const { height, width } = view;
  ...
```
And then remove the check on `currViewState` in `changed?

Reply:
I think this is more explicit since the `changed` is exactly what it is called i.e a list of changed views, not those that are changed and those that have a current `view`

Reply:
I refactored that method. While `props.viewStates` is usually small, I don't really like the multiple calls to `viewStates.find()`. The new implementation makes a shallow copy of `this.state.viewStates` and replaces entries for views where the view state changed. If anything changed, the state is updated.